module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -459,6 +450,15 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                 if ($item->discoverySuppress ?? false) {
                     continue;
                 }
+                $locationName = '';
+                if (!empty($item->effectiveLocationId)) {
+                    $locationResponse = $this->makeRequest(
```

Comment:
How likely is it that multiple items will have the same effective location ID? I would guess it's fairly probable. Would it make sense to cache location names locally to prevent duplicate API calls? Maybe a protected getLocationName() method would be helpful.

Also, is effectiveLocationId guaranteed to be defined, or do we need a fallback clause to the old permanentLocationId method?

Reply:
Thanks Demian, yes caching would be a really good idea here. I'll give it a try. My understanding is that effectiveLocationId is derived from the permanent/temporary Id fields and should always be available.

Reply:
@ihardy For example of caching you might want to take a look at the KohaRest driver here: https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/ILS/Driver/KohaRest.php#L1905. It needs a bit of setting up to with `use \VuFind\ILS\Driver\CacheTrait;` and setting up `$this->sessionCache` in init method, but after that it should be pretty straight-forward to use.

Reply:
Thanks for the example @EreMaijala that's very helpful. I gave this a try.

Reply:
Looks like a good start, @ihardy. Just two things to think about:

1.) You may want to override the getCacheKey method to ensure that multiple FOLIO drivers don't share the same cache values. Since VuFind supports union catalog scenarios, it's possible there could be more than one instance of the driver, and so each needs to provide its own unique cache keys. See https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/ILS/Driver/VoyagerRestful.php#L53-L55 and https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/ILS/Driver/VoyagerRestful.php#L331-L345 for how this is handled in the VoyagerRestful driver; I imagine FOLIO could do something similar based on the Okapi base URL.

2.) It might be worth adding a safety mechanism to refresh the cache if an expected identifier is not found in getLocationName (or to do a direct lookup of the offending ID), to cover the situation where a location gets added to the system and then looked up immediately before the cache has a chance to expire.